app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt

```diff
@@ -0,0 +1,312 @@
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EventLoggingConfigurationModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [IntroFragmentTest]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = IntroFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class IntroFragmentTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var context: Context
+
+  private val testProfileNickname = "John"
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  @Test
+  fun testFragment_explanationText_isDisplayed() {
+    launchOnboardingLearnerIntroActivity().use {
+      onView(withId(R.id.onboarding_learner_intro_title))
+        .check(matches(withText("Welcome, John!")))
+      onView(withText(R.string.onboarding_learner_intro_classroom_text))
+        .check(matches(isDisplayed()))
+      onView(withText(R.string.onboarding_learner_intro_practice_text))
+        .check(matches(isDisplayed()))
+      onView(
+        withText(
+          context.getString(
+            R.string.onboarding_learner_intro_feedback_text,
+            context.getString(R.string.app_name)
+          )
+        )
+      ).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFragment_portraitMode_stepCountText_isDisplayed() {
+    launchOnboardingLearnerIntroActivity().use {
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(withText(R.string.onboarding_step_count_four)))
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Testing lifecycle fails on Robolectric.
```

Comment:
Ditto for these tests (can be made to work on Robolectric).

Reply:
Done